File: src/main/java/org/jdc/template/domain/main/phonelistview/PhoneListView.java
/*
 * PhoneListView.java
 *
 * Created: 04/01/2014 10:46:44
 */



package org.jdc.template.domain.main.phonelistview;

import android.database.Cursor;
import android.content.ContentValues;

import org.jdc.template.domain.main.individual.Individual;

public class PhoneListView extends PhoneListViewBaseRecord {

    public static final String DROP_VIEW = "DROP VIEW IF EXISTS " + PhoneListView.TABLE + ";";
    public static final String CREATE_VIEW = "CREATE VIEW IF NOT EXISTS " + PhoneListView.TABLE + " AS SELECT " +
            Individual.FULL_C_ID + " AS " + PhoneListView.C_ID + ", " +
            Individual.FULL_C_LAST_NAME + " AS " + PhoneListView.C_NAME +
            " FROM " + Individual.TABLE;

    public PhoneListView(Cursor cursor) {
        setContent(cursor);
    }

    public PhoneListView(ContentValues values) {
        setContent(values);
    }

    public PhoneListView() {
    }

    public String getDropSql() {
        return DROP_VIEW;
    }

    public String getCreateSql() {
        return CREATE_VIEW;
    }


}

File: src/main/kotlin/org/jdc/template/domain/main/phonelistview/PhoneListView.kt
/*
 * PhoneListView.kt
 *
 * Created: 02/14/2016 07:02:10
 */



package org.jdc.template.domain.main.phonelistview

import android.content.ContentValues
import android.database.Cursor
import org.jdc.template.domain.main.individual.IndividualConst

class PhoneListView : PhoneListViewBaseRecord {

    companion object {
        val DROP_VIEW: String
        val CREATE_VIEW: String

        init {
            DROP_VIEW = "DROP VIEW IF EXISTS " + PhoneListViewConst.TABLE + ";";
            CREATE_VIEW = "CREATE VIEW IF NOT EXISTS " + PhoneListViewConst.TABLE + " AS SELECT " +
                    IndividualConst.FULL_C_ID + " AS " + PhoneListViewConst.C_ID + ", " +
                    IndividualConst.FULL_C_LAST_NAME + " AS " + PhoneListViewConst.C_NAME +
                    " FROM " + IndividualConst.TABLE;
        }
    }

    constructor() {
    }

    constructor(cursor: Cursor) {
        setContent(cursor)
    }

    constructor(values: ContentValues) {
        setContent(values)
    }

    fun getDropSql(): String {
        return DROP_VIEW
    }

    fun getCreateSql(): String {
        return CREATE_VIEW
    }


}